### PR TITLE
[CHK-3458] remapping npg/redirect outcome for app io outcome

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Depends on https://github.com/pagopa/pagopa-infra/pull/2605
In this PR ecommerce-fe outcome for NPG payment flows and redirect payment flow is updated. 
More over also some polling statuses are updated in order to reduce GENERIC_ERROR page.

#### List of Changes

- prefer PSP_ERROR(25) versus GENERIC_ERROR(1) in final polling status evaluation
- use NPG error code to detect some specific auth issue (cvv error, balance limit, payment method limit)
- remap some redirect status to PSP_ERROR(25)
- update infra commit refs

#### Motivation and Context

The goal is to improve user experience displaying a more readable error content

#### How Has This Been Tested?

- local tests
- dev tests
- integration tests
- unit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/api/transactions/types.ts [hunk 1]
```diff
@@ -17,6 +17,11 @@ export enum ViewOutcomeEnum {
   CANCELED_BY_USER = "8",
   EXCESSIVE_AMOUNT = "10",
   TAKING_CHARGE = "17",
+  PSP_ERROR = "25",
+  BE_KO = "99",
+  BALANCE_LIMIT = "116",
+  CVV_ERROR = "117",
+  LIMIT_EXCEDEED = "121",
```
Comment (scaminati-bv) [inline src/utils/api/transactions/types.ts:24]:
```suggestion
  LIMIT_EXCEEDED = "121",
```

**Review by CianoDanilo — APPROVED** [review]

File: src/utils/api/transactions/types.ts (hunk 1 above)
Comment (ciuffagianluca) [reply src/utils/api/transactions/types.ts]: fixed in 4d0be32a7f86afd87ad04200d4b1063eded644d7

**Review by scaminati-bv — APPROVED** [review]

File: src/utils/api/transactions/TransactionResultUtil.ts [hunk 2]
```diff
@@ -74,46 +76,73 @@ export const getOnboardingPaymentOutcome = (
...
+        wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
+      ) {
+        return ViewOutcomeEnum.PSP_ERROR;
+      } else if (
+        sendPaymentResultOutcome ===
+          SendPaymentResultOutcomeEnum.NOT_RECEIVED &&
+        wasAuthorizedByGateway(gateway, gatewayAuthorizationStatus)
+      ) {
+        return ViewOutcomeEnum.TAKING_CHARGE;
       }
```
Comment (scaminati-bv) [inline src/utils/api/transactions/TransactionResultUtil.ts:146]: How about using a switch case?

```suggestion
     } else {
          switch (sendPaymentResultOutcome) {
            case SendPaymentResultOutcomeEnum.OK:
              return ViewOutcomeEnum.SUCCESS;
            case SendPaymentResultOutcomeEnum.KO:
              return ViewOutcomeEnum.PSP_ERROR;
            case SendPaymentResultOutcomeEnum.NOT_RECEIVED:
              return ViewOutcomeEnum.TAKING_CHARGE;
            default:
              return ViewOutcomeEnum.GENERIC_ERROR; // BE_KO(99)
          }
     }
```

**Review by scaminati-bv — APPROVED** [review]

**Review by EmanueleBVtech — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -4,8 +4,8 @@
...
-    "generate:payment-ecommerce": "rimraf generated/definitions/payment-ecommerce && shx mkdir -p generated/definitions/payment-ecommerce && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/v1.200.0/src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl --no-strict --out-dir ./generated/definitions/payment-ecommerce --request-types --response-decoders --client",
+    "generate:payment-ecommerce-webview": "rimraf generated/definitions/payment-ecommerce-webview && shx mkdir -p generated/definitions/payment-ecommerce-webview && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/c268b2f036d55c05b5740060a8849887a549c74e/src/domains/ecommerce-app/api/ecommerce-webview/v1/_openapi.json.tpl --no-strict --out-dir ./generated/definitions/payment-ecommerce-webview --request-types --response-decoders --client",
```
Comment (pietro-tota) [inline package.json:7]: TODO: update infra ref after dependent pr merge

File: src/utils/api/transactions/TransactionResultUtil.ts [hunk 4]
```diff
@@ -46,19 +46,21 @@ export const gatewayAuthorizationStatusMap = new Map<
...
+  ["909", ViewOutcomeEnum.PSP_ERROR],
+  ["911", ViewOutcomeEnum.PSP_ERROR],
+  ["913", ViewOutcomeEnum.PSP_ERROR],
+  ["999", ViewOutcomeEnum.PSP_ERROR],
 ]);
 
 // eslint-disable-next-line complexity
 export const getOnboardingPaymentOutcome = (
   transactionInfo: transactionInfoStatus
+  // to be removed when also redirect gateway mapping will be set to PSP_ERROR in place of GENERIC_ERROR
```
Comment (pietro-tota) [inline src/utils/api/transactions/TransactionResultUtil.ts:62]: forgotten TODO?

**Review by pietro-tota — APPROVED** [review]
LGTM, just some trivial issues

File: package.json (hunk 3 above)
Comment (ciuffagianluca) [reply package.json]: Done in 6ed05ea880e7e4264f2f5c39e4be75ee0c6d10ef

**Pull request merged**